packages/main/src/TimePickerInternals.ts

```diff
@@ -0,0 +1,440 @@
+import UI5Element from "@ui5/webcomponents-base/dist/UI5Element.js";
+import customElement from "@ui5/webcomponents-base/dist/decorators/customElement.js";
+import property from "@ui5/webcomponents-base/dist/decorators/property.js";
+import event from "@ui5/webcomponents-base/dist/decorators/event.js";
+import { getI18nBundle } from "@ui5/webcomponents-base/dist/i18nBundle.js";
+import type I18nBundle from "@ui5/webcomponents-base/dist/i18nBundle.js";
+import getLocale from "@ui5/webcomponents-base/dist/locale/getLocale.js";
+import DateFormat from "@ui5/webcomponents-localization/dist/DateFormat.js";
+import getCachedLocaleDataInstance from "@ui5/webcomponents-localization/dist/getCachedLocaleDataInstance.js";
+import "@ui5/webcomponents-localization/dist/features/calendar/Gregorian.js"; // default calendar for bundling
+import CalendarType from "@ui5/webcomponents-base/dist/types/CalendarType.js";
+import { fetchCldr } from "@ui5/webcomponents-base/dist/asset-registries/LocaleData.js";
+import Integer from "@ui5/webcomponents-base/dist/types/Integer.js";
+import SegmentedButton from "./SegmentedButton.js";
+import {
+	getHoursConfigByFormat,
+	getTimeControlsByFormat,
+	HourType,
+} from "./timepicker-utils/TimeSlider.js";
+import {
+	TIMEPICKER_HOURS_LABEL,
+	TIMEPICKER_MINUTES_LABEL,
+	TIMEPICKER_SECONDS_LABEL,
+	TIMEPICKER_CLOCK_DIAL_LABEL,
+} from "./generated/i18n/i18n-defaults.js";
+
+type TimePickerComponentIndexMap = {
+	hours: number,
+	minutes: number,
+	seconds: number,
+}
+
+type TimeSelectionPeriodProperties = {
+	label: string,
+	pressed: boolean,
+}
+
+type TimeSelectionChangeEventDetail = {
+	value: string | undefined,
+	valid: boolean,
+}
+
+/**
+ * @class
+ *
+ * <h3 class="comment-api-title">Overview</h3>
+ *
+ * <code>ui5-time-picker-internals</code> is helper component that contains shared methods used in <code>ui5-time-selection-clocks</code>
+ * and <ui5-time-selection-inputs> components and should not be used separately.
```

Comment:
- what is `ui5-time-selection-inputs`, is it used 
- is `ui5-time-picker-internals` used somewhere or this is a base class

Reply:
The mentioned component (ui5-time-selection-inputs) will be developed and added to the timepicker as a next step - it is a part of the redesign. It will affect only the usage on mobile devices, giving a separate popover with numeric inputs, where simple numeric keyboard will be displayed instead of full one. As the big part of the functionality used by ui5-time-selection-clocks will be used in ui5-time-selection-inputs too, it is separated in private ui5-time-picker-internals component.